src/lt/plugins/auto_complete.cljs

```diff
@@ -148,20 +148,34 @@
 
 (declare hinter)
 
+(defn remove-long-completitions [hints]
+  (filter #(< (.-length (.-completion %)) (:hint-limit @hinter)) hints))
+
 (def hinter (-> (scmd/filter-list {:items (fn []
                                             (when-let [cur (pool/last-active)]
                                               (let [token (-> @hinter :starting-token :string)]
                                                 (distinct-completions
```

Comment:
I think with yet another level of nesting perhaps a threading would read better, maybe something along the lines of:
```clojure
(->> (object/raise-reduce cur :hints+ [] token)
     (remove #(= token (.-completion %)))
     remove-long-completitions
     distinct-completions)```